### PR TITLE
[ADD]open_academy: Added reports,wizard and i18n T#60931

### PR DESCRIPTION
**Pull request by UlisesRivadeneyra**
Added the folder i18n with files inside that are diferent lenguages for the odoo modules 
-es.po

Added reports folder that contains an xml to generate a report file and view in odoo 
-openacademy_session_reports.xml

Added wizard folder that contain an odoo wizard pop up in sessions model 
-open_academy_wizard_views.xml
-open_academy_wizard.py

### CODE REVIEW

**Comment by UlisesRivadeneyra** [comment]
Hi @edy1192 , mi pr is ready to be reviewed with all exercises of buiding a module

File: open_academy/i18n/es.po [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+"                                Fecha de inicio:"
+
+#. module: open_academy
+#: model_terms:ir.ui.view,arch_db:open_academy.view_openacad_session_kanban
+msgid ""
+"<br/>\n"
+"                                duration:"
+msgstr ""
+"<br/>\n"
+"                                duracion:"
```
Comment (edy1192) [inline open_academy/i18n/es.po:34]:
```suggestion
"                                duración:"
```

File: open_academy/i18n/es.po [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+#. module: open_academy
+#: model_terms:ir.ui.view,arch_db:open_academy.report_session_view
+msgid "<strong>Start date:</strong>"
+msgstr "<strong>Fecha de inicio:</strong>"
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session__active
+msgid "Active"
+msgstr ""
```
Comment (edy1192) [inline open_academy/i18n/es.po:59]:
```suggestion
msgstr "Activo"
```

File: open_academy/i18n/es.po [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+
+#. module: open_academy
+#: model_terms:ir.ui.view,arch_db:open_academy.wizard_form
+msgid "Add Attendees"
+msgstr "Añadir asistente"
```
Comment (edy1192) [inline open_academy/i18n/es.po:64]:
```suggestion
msgstr "Añadir asistentes"
```

File: open_academy/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_course__description
+msgid "Description of the course"
+msgstr "Descripción de curso"
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_course__display_name
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session__display_name
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_wizard__display_name
+msgid "Display Name"
+msgstr "Nombre para mostrar"
```
Comment (edy1192) [inline open_academy/i18n/es.po:157]: Standard is:
```suggestion
msgstr "Nombre mostrado"
```

File: open_academy/i18n/es.po [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session__end_date
+msgid "End Date"
+msgstr "Fecha de finalización"
+
+#. module: open_academy
+#: model_terms:ir.ui.view,arch_db:open_academy.openacademy_course_search
+msgid "Group By"
+msgstr "Agrupar por"
```
Comment (edy1192) [inline open_academy/i18n/es.po:177]:
```suggestion
msgstr "Agrupado por"
```

File: open_academy/i18n/es.po [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_course__id
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session__id
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_wizard__id
+msgid "ID"
+msgstr "ID"
```
Comment (edy1192) [inline open_academy/i18n/es.po:184]: When both terms are the same, no translation is added.

```suggestion
msgstr ""
```

File: open_academy/i18n/es.po [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session__instructor_id
+#: model:ir.model.fields,field_description:open_academy.field_res_partner__instructor
+#: model:ir.model.fields,field_description:open_academy.field_res_users__instructor
+msgid "Instructor"
+msgstr "Instructor"
```
Comment (edy1192) [inline open_academy/i18n/es.po:191]: When both terms are the same, no translation is added.

```suggestion
msgstr ""
```

File: open_academy/i18n/es.po [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_course____last_update
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session____last_update
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_wizard____last_update
+msgid "Last Modified on"
+msgstr "Última vez modificado por"
```
Comment (edy1192) [inline open_academy/i18n/es.po:198]: Standard is:
```suggestion
msgstr "Última modificación el"
```

File: open_academy/i18n/es.po [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_course__write_uid
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session__write_uid
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_wizard__write_uid
+msgid "Last Updated by"
+msgstr "Última vez actualizado por"
```
Comment (edy1192) [inline open_academy/i18n/es.po:205]: Standard is:
```suggestion
msgstr "Última actualización por"
```

File: open_academy/i18n/es.po [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_course__write_date
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session__write_date
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_wizard__write_date
+msgid "Last Updated on"
+msgstr "Última actualización en"
```
Comment (edy1192) [inline open_academy/i18n/es.po:212]:
```suggestion
msgstr "Última actualización el"
```

File: open_academy/i18n/es.po [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+#: model_terms:ir.ui.view,arch_db:open_academy.open_academy_calendar_view
+msgid "Session Calendar"
+msgstr "Calendario de la sesión"
+
+#. module: open_academy
+#: model:ir.actions.act_window,name:open_academy.open_board_session
+#: model:ir.ui.menu,name:open_academy.menu_board_session
+#: model_terms:ir.ui.view,arch_db:open_academy.board_session_form
+msgid "Session Dashboard"
+msgstr "Pizarron de la sesión"
```
Comment (edy1192) [inline open_academy/i18n/es.po:272]:
```suggestion
msgstr "Tablero de la sesión"
```

File: open_academy/i18n/es.po [hunk 12]
```diff
@@ -0,0 +1,337 @@
...
+
+#. module: open_academy
+#: model:ir.actions.act_window,name:open_academy.openacademy_wizard_action
+msgid "Session actions"
+msgstr "Acciones de la sesión"
+
+#. module: open_academy
+#: model_terms:ir.ui.view,arch_db:open_academy.view_openacad_session_kanban
+msgid "Session name:"
+msgstr "NOmbre de la sesión"
```
Comment (edy1192) [inline open_academy/i18n/es.po:292]:
```suggestion
msgstr "Nombre de la sesión"
```

File: open_academy/models/open_academy_partner.py [hunk 13]
```diff
@@ -0,0 +1,8 @@
+from odoo import models, fields
```
Comment (edy1192) [inline open_academy/models/open_academy_partner.py:1]: Rename this file check the file in main branch https://github.com/UlisesRivadeneyra/Open-Academy/blob/main/open_academy/models/res_partner.py

File: open_academy/models/open_academy_partner.py [hunk 14]
```diff
@@ -0,0 +1,8 @@
...
+
+
+class Partner(models.Model):
```
Comment (edy1192) [inline open_academy/models/open_academy_partner.py:4]: Fix this https://github.com/UlisesRivadeneyra/Open-Academy/blob/main/open_academy/models/res_partner.py#L4

File: open_academy/models/open_academy_session.py [hunk 15]
```diff
@@ -79,7 +79,7 @@ def _compute_taken_seats(self):
     def _check_instructor_not_in_attendees(self):
         for record in self.filtered('instructor_id'):
             if record.instructor_id in record.attendee_ids:
-                raise exceptions.ValidationError("A session's instructor can't be an attendee.")
+                raise exceptions.ValidationError_("A session's instructor can't be an attendee.")
```
Comment (edy1192) [inline open_academy/models/open_academy_session.py:82]: I don't see the translation for this text.

File: open_academy/views/open_academy_course_views.xml [hunk 16]
```diff
@@ -15,6 +15,9 @@
                 <sheet>
                     <group>
                         <field name="title"></field>
+                        <field name="description"></field>
+                        <field name="responsible_user_id"></field>
+                        <field name="session_ids"></field>
```
Comment (edy1192) [inline open_academy/views/open_academy_course_views.xml:20]: You duplicate this fields.

File: open_academy/views/open_academy_menu_views.xml [hunk 17]
```diff
@@ -5,4 +5,6 @@
     parent='open_academy_root_menu' action="openacademy_course_action_list"/>
     <menuitem id="open_academy_session_menu" name="Sessions" sequence="2"
     parent='open_academy_root_menu' action="session_list_action"/>
+    <menuitem id="menu_board_session" name="Session Dashboard" parent="base.menu_reporting_dashboard"
```
Comment (edy1192) [inline open_academy/views/open_academy_menu_views.xml:8]: Why this parent? Use the open_academy_root_menu
```suggestion
    <menuitem id="menu_board_session" name="Session Dashboard" parent="base.menu_reporting_dashboard"
```

File: open_academy/views/open_academy_partner_views.xml [hunk 18]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (edy1192) [inline open_academy/views/open_academy_partner_views.xml:1]: Check this file in main branch https://github.com/UlisesRivadeneyra/Open-Academy/blob/main/open_academy/views/res_partner_views.xml

File: open_academy/wizard/open_academy_wizard.py [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class Wizard(models.TransientModel):
+    _name = 'openacademy.wizard'
+    _description = 'Register attendees'
+
+    attendee_ids = fields.Many2many('res.partner', string="Attendees")
+
+    def _default_session(self):
+        return self.env['openacademy.session'].browse(self._context.get('active_id'))
```
Comment (edy1192) [inline open_academy/wizard/open_academy_wizard.py:11]: This method should be after fields see guidelines https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#37models
```suggestion
    def _default_session_id(self):
```

When you move it you need to change the field and use lambda function in the default parameter to find the method.
`default=lambda self: self._default_session_id()`

File: open_academy/wizard/open_academy_wizard.py [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+    session_id = fields.Many2one(
+        'openacademy.session',
+        required=True,
+        default=_default_session
+    )
+
+    def subscribe(self):
+        for session in self.session_ids:
+            session.attendee_ids |= self.attendee_ids
+        return {}
```
Comment (edy1192) [inline open_academy/wizard/open_academy_wizard.py:22]: This is not necessary.

```suggestion
```

File: open_academy/wizard/open_academy_wizard_views.xml [hunk 21]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="wizard_form" model="ir.ui.view" >
+        <field name="name">wizard.form</field>
+        <field name="model">openacademy.wizard</field>
+        <field name="arch" type="xml">
+            <form string="Add Attendees">
+                <group>
+                    <field name="session_id" widget="many2many_list"/>
```
Comment (edy1192) [inline open_academy/wizard/open_academy_wizard_views.xml:9]: I think this widget is  not necessary.

```suggestion
                    <field name="session_id"/>
```

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by edy1192** [comment]
Fix the PR's description please.

**Comment by UlisesRivadeneyra** [comment]
hi @edy1192 i modify the code that you put me in the comment
Cheers, Ulises

File: open_academy/i18n/es.po [hunk 22]
```diff
@@ -0,0 +1,337 @@
...
+#: model:ir.model.fields,field_description:open_academy.field_res_users__instructor
+msgid "Instructor"
+msgstr ""
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_course____last_update
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session____last_update
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_wizard____last_update
+msgid "Last Modified on"
+msgstr "Última vez modificado el"
```
Comment (edy1192) [inline open_academy/i18n/es.po:198]: Pending https://github.com/UlisesRivadeneyra/Open-Academy/pull/8#discussion_r996069553

File: open_academy/models/open_academy_session.py (hunk 15 above)
Comment (edy1192) [reply open_academy/models/open_academy_session.py:82]: Pending

**Review by edy1192 — CHANGES_REQUESTED** [review]

File: open_academy/models/open_academy_session.py (hunk 15 above)
Comment (edy1192) [reply open_academy/models/open_academy_session.py:82]: Add translations to onchange's message

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by UlisesRivadeneyra** [comment]
@edy1192 i already change the comments that you give me with the new translations.

File: open_academy/i18n/es.po [hunk 23]
```diff
@@ -0,0 +1,367 @@
...
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_session__id
+#: model:ir.model.fields,field_description:open_academy.field_openacademy_wizard__id
+msgid "ID"
+msgstr ""
+
+#. module: open_academy
+#: code:addons/open_academy/models/open_academy_session.py:0
+#, python-format
+msgid "Incorrect 'seats' value."
+msgstr ""
```
Comment (edy1192) [inline open_academy/i18n/es.po:196]: Missing translation.

**Review by edy1192 — APPROVED** [review]
Apply my comment in your next PR please.

**Review by edy1192 — APPROVED** [review]

**Pull request merged**